src/ngraph/op/strided_slice.cpp

```diff
@@ -0,0 +1,191 @@
+                                   const std::vector<int64_t>& end_mask,
+                                   const std::vector<int64_t>& new_axis_mask,
+                                   const std::vector<int64_t>& shrink_axis_mask,
+                                   const std::vector<int64_t>& ellipsis_mask)
+    : Op({data, begin, end, strides})
+    , m_begin_mask{begin_mask}
+    , m_end_mask{end_mask}
+    , m_new_axis_mask{new_axis_mask}
+    , m_shrink_axis_mask{shrink_axis_mask}
+    , m_ellipsis_mask{ellipsis_mask}
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v1::StridedSlice::StridedSlice(const Output<Node>& data,
+                                   const Output<Node>& begin,
+                                   const Output<Node>& end,
+                                   const std::vector<int64_t>& begin_mask,
+                                   const std::vector<int64_t>& end_mask,
+                                   const std::vector<int64_t>& new_axis_mask,
+                                   const std::vector<int64_t>& shrink_axis_mask,
+                                   const std::vector<int64_t>& ellipsis_mask)
+    : StridedSlice(data,
+                   begin,
+                   end,
+                   op::Constant::create(element::i64,
+                                        Shape{begin_mask.size()},
+                                        vector<int64_t>(begin_mask.size(), 1)),
+                   begin_mask,
+                   end_mask,
+                   new_axis_mask,
+                   shrink_axis_mask,
+                   ellipsis_mask)
+{
+}
+
+void op::v1::StridedSlice::validate_and_infer_types()
+{
+    const auto& begin_mask_et = get_input_element_type(1);
+    const auto& end_mask_et = get_input_element_type(2);
+    NODE_VALIDATION_CHECK(this,
+                          begin_mask_et.compatible(element::Type_t::i64),
+                          "Begin mask must have element type i64, but has ",
+                          begin_mask_et);
+    NODE_VALIDATION_CHECK(this,
+                          end_mask_et.compatible(element::Type_t::i64),
+                          "End mask must have element type i64, but has ",
+                          end_mask_et);
+
+    auto are_mask_elem_in_range = [](size_t e) { return e == 0 || e == 1; };
+    NODE_VALIDATION_CHECK(
+        this,
+        std::all_of(m_begin_mask.begin(), m_begin_mask.end(), are_mask_elem_in_range) &&
+            std::all_of(m_end_mask.begin(), m_end_mask.end(), are_mask_elem_in_range) &&
+            std::all_of(m_new_axis_mask.begin(), m_new_axis_mask.end(), are_mask_elem_in_range) &&
+            std::all_of(
+                m_shrink_axis_mask.begin(), m_shrink_axis_mask.end(), are_mask_elem_in_range) &&
+            std::all_of(m_ellipsis_mask.begin(), m_ellipsis_mask.end(), are_mask_elem_in_range),
+        "All maks elements of StridedSlice must have be 0 or 1");
```

Comment:
minor typo: maks -> mask 
same in other places

Reply:
Fixed